beacon-chain/rpc/aggregator/server.go

```diff
@@ -0,0 +1,85 @@
+package aggregator
+
+import (
+	"context"
+	"errors"
+
+	"github.com/prysmaticlabs/prysm/beacon-chain/blockchain"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/state"
+	"github.com/prysmaticlabs/prysm/beacon-chain/db"
+	"github.com/prysmaticlabs/prysm/beacon-chain/sync"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/rpc/v1"
+	"github.com/prysmaticlabs/prysm/shared/bls"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/sirupsen/logrus"
+	"go.opencensus.io/trace"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var log logrus.FieldLogger
+
+func init() {
+	log = logrus.WithField("prefix", "rpc/aggregator")
+}
+
+// Server defines a server implementation of the gRPC aggregator service.
+type Server struct {
+	BeaconDB    db.Database
+	HeadFetcher blockchain.HeadFetcher
+	SyncChecker sync.Checker
+}
+
+// SubmitAggregateAndProof is called by a validator at every slot to check whether
```

Comment:
Every slot? Is there no way to determine this in assignments?

Reply:
Updated comment. Should be every slot if you are part of the committee